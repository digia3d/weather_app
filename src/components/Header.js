import React from 'react';
import { FaAngleLeft } from 'react-icons/fa';
import { Link } from 'react-router-dom';

const dateBuilder = (d) => {
  const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
  const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];

  const day = days[d.getDay()];
  const date = d.getDate();
  const month = months[d.getMonth()];
  const year = d.getFullYear();

  return `${day} ${date} ${month} ${year}`;
};

function Header() {
  return (
    <main>
      <header className="header">
        <div className="logo">
          <Link className="logo-link" to="/Home">
            <h2>
              <FaAngleLeft size={24} />
              Back
            </h2>
          </Link>
          <Link className="logo-link" to="/City">City</Link>
        </div>
        <div className="container">
          <div className="title">
            <h1>Weather App</h1>
          </div>
          <div className="date">
            {dateBuilder(new Date())}
          </div>
        </div>
      </header>
    </main>
  );
}

export default Header;
