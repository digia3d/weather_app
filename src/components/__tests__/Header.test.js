import React from "react";
import renderer from 'react-test-renderer';
import { HashRouter as Router } from "react-router-dom";
import Header from "../Header";
import '@testing-library/jest-dom';

describe('Test of Header', () => {
  it('test if Header is rendered', () => {
      const headerTest = renderer
          .create(<Router><Header /></Router>)
          .toJSON();
      expect(headerTest).toMatchSnapshot();
  });
});