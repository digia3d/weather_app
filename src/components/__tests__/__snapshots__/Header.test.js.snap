// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Header test if Header is rendered 1`] = `
<main>
  <header
    className="header"
  >
    <a
      className="logo-link"
      href="#/Home"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height={24}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 256 512"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
    </a>
    <div
      className="date"
    >
      Saturday 3 September 2022
    </div>
  </header>
  <section
    className="section"
  >
    <div
      className="conttitl"
    >
      <p
        className="title"
      >
        Weather App
      </p>
      <h1
        className="titleM"
      >
        London
      </h1>
    </div>
  </section>
</main>
`;
